src/module.c

```diff
@@ -41,6 +41,19 @@
     sptmp;                                                                                  \
   })
 
+// Check if the current request can be executed in a threadb
+static int CheckConcurrentSupport(RedisModuleCtx *ctx) {
+  // See if this client should be concurrent
+  if (RSGlobalConfig.concurrentMode) {
+    return 0;
+  }
+  if (RedisModule_GetContextFlags && (RedisModule_GetContextFlags(ctx) &
```

Comment:
Please add comments documenting what this logic means and why we can't work concurrently in Lua and Multi